identity-did/src/document/core_document.rs

```diff
@@ -521,79 +521,45 @@ where
-        self.authentication.query(query.into()).and_then(resolve_ref_helper)
-      }
-      MethodScope::VerificationRelationship(MethodRelationship::AssertionMethod) => {
-        self.assertion_method.query(query.into()).and_then(resolve_ref_helper)
-      }
-      MethodScope::VerificationRelationship(MethodRelationship::KeyAgreement) => {
-        self.key_agreement.query(query.into()).and_then(resolve_ref_helper)
+      Some(scope) => {
+        let resolve_ref_helper = |method_ref: &'me MethodRef<D, U>| self.resolve_method_ref(method_ref);
+
+        match scope {
+          MethodScope::VerificationMethod => self.verification_method.query(query.into()),
+          MethodScope::VerificationRelationship(MethodRelationship::Authentication) => {
+            self.authentication.query(query.into()).and_then(resolve_ref_helper)
+          }
+          MethodScope::VerificationRelationship(MethodRelationship::AssertionMethod) => {
+            self.assertion_method.query(query.into()).and_then(resolve_ref_helper)
+          }
+          MethodScope::VerificationRelationship(MethodRelationship::KeyAgreement) => {
+            self.key_agreement.query(query.into()).and_then(resolve_ref_helper)
+          }
+          MethodScope::VerificationRelationship(MethodRelationship::CapabilityDelegation) => self
+            .capability_delegation
+            .query(query.into())
+            .and_then(resolve_ref_helper),
+          MethodScope::VerificationRelationship(MethodRelationship::CapabilityInvocation) => self
+            .capability_invocation
+            .query(query.into())
+            .and_then(resolve_ref_helper),
+        }
       }
-      MethodScope::VerificationRelationship(MethodRelationship::CapabilityDelegation) => self
-        .capability_delegation
-        .query(query.into())
-        .and_then(resolve_ref_helper),
-      MethodScope::VerificationRelationship(MethodRelationship::CapabilityInvocation) => self
-        .capability_invocation
-        .query(query.into())
-        .and_then(resolve_ref_helper),
+      None => self.resolve_method_inner(query.into()),
     }
   }
 
-  /// Returns the first [`VerificationMethod`] with an `id` property matching the provided `query`
-  /// and the verification relationship specified by `scope`.
-  ///
-  /// # Errors
-  ///
-  /// Fails if no matching [`VerificationMethod`] is found.
-  pub fn try_resolve_method_with_scope<'query, 's: 'query, Q>(
-    &'s self,
-    query: Q,
-    scope: MethodScope,
-  ) -> Result<&VerificationMethod<D, U>>
-  where
-    Q: Into<DIDUrlQuery<'query>>,
-  {
-    self
-      .resolve_method_with_scope(query, scope)
-      .ok_or(Error::MethodNotFound)
-  }
-
   /// Returns a mutable reference to the first [`VerificationMethod`] with an `id` property
   /// matching the provided `query`.
   pub fn resolve_method_mut<'query, Q>(&mut self, query: Q) -> Option<&mut VerificationMethod<D, U>>
```

Comment:
`CoreDocument::resolve_method_mut` should also be refactored to take `Option<MethodScope>`.